include/openrave/collisionchecker.h

```diff
@@ -73,6 +73,8 @@ class OPENRAVE_API CollisionReport
         Reset();
     }
 
+    virtual ~CollisionReport() noexcept {}
```

Comment:
@undisputed-seraphim just curious, what is the purpose of adding this here?

Reply:
A class that has methods marked `virtual` should have also have a `virtual` destructor. [See this.](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-virtual-class-destructor.html) This is so that so class instances can be destroyed properly when accessing them through pointers.

It's just a warning, but since this is in a header file, the warning is emitted once every time the header is encountered, which is many times, and adds a lot of noise to compilation.